### PR TITLE
Fix(oracle): allow CONNECT BY / START WITH to be interchanged

### LINKED ISSUE #2205: Unable to parse the oracle query when query has CONNECT BY followed by START WITH clause
Hello Team,
Warm greetings!

Parser is throwing an error when there is CONNECT BY followed by START WITH clause!

**Oracle Documentation -**
https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/Hierarchical-Queries.html#GUID-0118DF1D-B9A9-41EB-8556-C6E7D6A5A84E
image

**Code for your reference -**
```

import sqlglot.expressions as exp

# works fine
query1 ="""
SELECT last_name "Employee", 
   LEVEL, SYS_CONNECT_BY_PATH(last_name, '/') "Path"
   FROM employees
   WHERE level <= 3 AND department_id = 80
   START WITH last_name = 'King'
   CONNECT BY PRIOR employee_id = manager_id AND LEVEL <= 4
"""
# Doesn't work
query2 ="""
SELECT last_name "Employee", 
   LEVEL, SYS_CONNECT_BY_PATH(last_name, '/') "Path"
   FROM employees
   WHERE level <= 3 AND department_id = 80
   CONNECT BY PRIOR employee_id = manager_id AND LEVEL <= 4
   START WITH last_name = 'King'
"""

expressions = sqlglot.parse_one(query2, read='oracle')
x = list(expressions.find_all(exp.Table))

for i in x:
  listTables = str(i).split(' ')[0]
  print(listTables)
```

Note that above queries are valid oracle queries.

query2 fails as START WITH clause is after the connect by clause.
Request you to add support for the same.

Thanks ton in advance!

Thanks,
Sachin

### PR DESCRIPTION
**Pull request by GeorgeSittas**
Fixes #2205

### CODE REVIEW

File: sqlglot/parser.py [hunk 1]
```diff
@@ -2913,6 +2913,10 @@ def _parse_connect(self, skip_start_token: bool = False) -> t.Optional[exp.Conne
         )
         connect = self._parse_conjunction()
         self.NO_PAREN_FUNCTION_PARSERS.pop("PRIOR")
+
+        if self._match(TokenType.START_WITH):
```
Comment (tobymao) [inline sqlglot/parser.py:2917]: if not start and ...

Comment (GeorgeSittas) [reply sqlglot/parser.py:2917]: this check should be redundant right?

Comment (tobymao) [reply sqlglot/parser.py:2917]: well, then you'd allow something like

```start by x connect by y start by z```

Comment (GeorgeSittas) [reply sqlglot/parser.py:2917]: k i'll change this real quick

**Pull request merged**